src/components/text-area/TextArea.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { forwardRef, useMemo, useRef, useState } from "react";
+import { Platform, TextInputProps, ViewStyle } from "react-native";
+
+import { Box, BoxProps, RNTextInput, TouchableProps } from "../../primitives";
+import { useTheme } from "../../theme";
+import {
+  cx,
+  generateBoxShadow,
+  RenderPropType,
+  runIfFn,
+  styleAdapter,
+  useOnFocus,
+  useOnHover,
+} from "../../utils";
+import { createComponent } from "../../utils/createComponent";
+import { composeEventHandlers } from "../../utils/mergeRefs";
+import { Spinner } from "../spinner";
+import { createIcon, Icon, TextAreaSuffix } from "..";
+
+export type TextAreaVariants = "outline" | "subtle" | "underline" | "ghost";
+export type TextAreaSizes = "sm" | "md" | "lg" | "xl";
+
+export interface TextAreaProps extends TextInputProps {
```

Comment:
I think we have not handled the `disabled` case.

Reply:
editable prop is there for the disabled case

Reply:
Oh, then add it as a controllable.

Reply:
<image src="https://user-images.githubusercontent.com/20360870/201653348-0b18c92d-166f-4ebc-a31b-5154873b3cd7.png" height="800" width="350" />

Reply:
Also added disabled state